static/src/javascripts/projects/commercial/modules/dfp/should-refresh.ts

```diff
@@ -0,0 +1,54 @@
+import type { AdSizeString } from '@guardian/commercial-core';
+import { adSizes } from '@guardian/commercial-core';
+import config from '../../../../lib/config';
+import type { Advert } from './Advert';
+
+const outstreamSizes = [
+	adSizes.outstreamDesktop.toString(),
+	adSizes.outstreamMobile.toString(),
+	adSizes.outstreamGoogleDesktop.toString(),
+];
+
+/**
+ * Determine whether an advert should refresh, taking into account
+ * its size, whether there's a pageskin or whether the advert's
+ * line item is marked as non-refreshable
+ *
+ *  - Fluid ads should not refresh
+ *  - Outstream ads should not refresh
+ *  - Pageskins should not refresh
+ *  - Ads that have line items marked as non-refreshable should not be
+ * 	  refreshed. This information is retrieved via the non refreshable
+ * 	  line item API endpoint
+ *
+ * @param advert The candidate advert to check
+ * @param nonRefreshableLineItemIds The array of line item ids for which
+ * adverts should not refresh
+ */
+export const shouldRefresh = (
+	advert: Advert,
+	nonRefreshableLineItemIds: number[] = [],
+): boolean => {
+	const sizeString = advert.size?.toString();
+
+	// Fluid adverts should not refresh
+	const isFluid = sizeString === '0,0';
+	if (isFluid) return false;
+
+	// Outstream adverts should not refresh
+	const isOutstream = outstreamSizes.includes(sizeString as AdSizeString);
+	if (isOutstream) return false;
+
+	// If the advert has a line item id included in the array of non refreshable
+	// line item ids then it should not refresh
+	const isNonRefreshableLineItem =
+		advert.lineItemId &&
+		nonRefreshableLineItemIds.includes(advert.lineItemId);
+	if (isNonRefreshableLineItem) return false;
+
+	// If we have a pageskin then don't refresh
+	if (config.get('page.hasPageSkin')) return false;
```

Comment:
Explicitly default to `false`?

```suggestion
	if (config.get<boolean>('page.hasPageSkin', false)) return false;
```

Or skip config.get altogether?

```suggestion
	if (window.guardian.config.page.hasPageSkin) return false;
```

Reply:
Definitely one for a subsequent PR.